### PR TITLE
Add RELEASE.md

### LINKED ISSUE #244: Create a RELEASE.md file that explains the website, doc, guides, and blogs release process'
#### Feature description

We need to write down the release process for our website, docs, guides, and blogs. Also what needs to happen if a hot fix needs to go into a current release.

# Website Release

The website is on its own release process. _add more explaining_

# Guides Release

Put this info in the draft-guide-template repo
There are no "releases" for guides just the removal of draft- gets them published on next website publish

# Blogs Release

on its own release process. For new blogs we will bump the **Minor** version and use the latest release when the website is published

# Docs Release

This is the release process for Kabanero as a whole - Kabanero uses [semver](https://semver.org/) versioning. The docs repo will follow this same process.

Here is an example flow of the Kabnero release using `0.2.0` as the next release example:

**Assumption**: When a new release of Kabanero as a whole is ready the next version number will always be a **Major** or **Minor** bump (never a **Patch** bump).

1. When 0.2.0 release date comes we cut the 0.2.0 release on the repo. End of release process.

## Docs repo needs a hot fix for 0.2.0 example

1. If after a release, A piece of Kabanero (cli, pipelines, operator, docs, etc...) needs to do a small update or a quick bug fix or something (and should be deployed before the next scheduled release, 0.3.0, in this example) we would:
1. create a `release-0.2` branch based off the 0.2.0 release. 
1. Put the changes in the `release-0.2` branch and `master` branch, if applicable (see Note below).
1. Cut a new release based off the `release-0.2` branch and bump the patch number so it becomes `0.2.1`
1. Publish release `0.2.1`.

**Note**: During this whole hox fit situation the `master` branch will always be going towards the next release so new stuff in master will always be `0.3.0` related.

### PR DESCRIPTION
**Pull request by kSee04**
#### What was fixed?  (Issue # or description of fix)
Closes #244

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+## Overview
+The content on the Kabanero.io Website is built from four different sources:
+- UI of the Website from: https://github.com/kabanero-io/kabanero-website
+- Docs from: https://github.com/kabanero-io/docs
+- Blogs from: https://github.com/kabanero-io/blogs
+- The guides from repositories in https://github.com/kabanero-io that begin with `guide-`
+
+## Website Release
+The website has its own independent release process.  Releases are versioned using standard semantic versioning of format **MAJOR.MINOR.PATCH**.
```
Comment (alohr51) [inline RELEASE.md:11]: Need to add _when_ we would do a release and what number gets bumped in the release number

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+
+## Guides Release
+There are no "releases" for the Kabanero.io Guides. As detailed in the [Kabanero.io Guides Contribution Guidelines](https://github.com/kabanero-io/draft-guides-template/blob/master/README.md), a repository containing a guide being drafted should begin with `draft-guide-`.  `draft-` ensures that the guide will not get published to the site.  Removing the `draft-` from the repo name so it begins with `guide-` ensures that our build process will pull in the guide during the next build.
```
Comment (alohr51) [inline RELEASE.md]: maybe add something to the second sentence like: "guides are an on/off situation. They are either published or not published depending on ...... "

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+
+## Blogs Release
+Blogs have their own independent release process.  
+
+Increment the **Minor** version when new blogs are added. The build will use the latest release when the website is published. 
+
+## Docs Release
+Docs follow the release process for Kabanero.io as a whole - Kabanero.io uses [semver](https://semver.org/) versioning.
```
Comment (alohr51) [inline RELEASE.md]: no `.io` here we're talking about Kabanero as a whole not the site

File: RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Here is an example flow of the Kabnero release using `0.2.0` as the next release example:
```
Comment (alohr51) [inline RELEASE.md]: typo on kabnero

File: RELEASE.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+**Assumption**: When a new release of Kabanero.io as a whole is ready the next version number will always be a **Major** or **Minor** bump (never a **Patch** bump).
```
Comment (alohr51) [inline RELEASE.md]: no .io here too

File: RELEASE.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+1. When 0.2.0 release date comes we cut the 0.2.0 release on the repo. End of release process.
```
Comment (alohr51) [inline RELEASE.md]: surround 0.2.0 in back ticks

File: RELEASE.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+### Docs repo needs a hot fix for 0.2.0 example
+
+1. If after a release, A piece of Kabanero.io (cli, pipelines, operator, docs, etc...) needs to do a small update or a quick bug fix or something (and should be deployed before the next scheduled release, 0.3.0, in this example) we would:
+1. create a `release-0.2` branch based off the 0.2.0 release. 
+1. Put the changes in the `release-0.2` branch and `master` branch, if applicable (see Note below).
+1. Cut a new release based off the `release-0.2` branch and bump the patch number so it becomes `0.2.1`
+1. Publish release `0.2.1`.
+
+**Note**: During this whole hox fit situation the `master` branch will always be going towards the next release so new stuff in master will always be `0.3.0` related.
```
Comment (alohr51) [inline RELEASE.md]: fit -> fix

**Review by alohr51 — APPROVED** [review]

**Review by garypicher — COMMENTED** [review]
Does semantic versioning make sense for a website?  In general, semantic versioning deals with software that exposes APIs, and defines rules around what number to increment (major / minor / patch) based on whether the APIs are changing, and whether they're changing in a backwards incompatible way.  I'm not sure that fits into what a website does.

Is the website content specific to a version of the rest of Kabanero?  If so, it might be worth keeping the website version in sync with kabanero-foundation.  If not, then maybe we should keep the website version in sync with the kabanero MAJOR version, but bump the minor version independently.  For example, keep the major version matching kabanero-foundation, but bump the website's minor version whenever we feel like it (whenever there's a need for content) and not expect it to match the minor version of kabanero-foundation.

I'm not sure what the right answer is, but some food for thought, anyway...

**Review by garypicher — APPROVED** [review]

**Comment by alohr51** [comment]
Yeah semantic versioning might be entirely useful since nothing depends on it, I guess its just a set of versioning rules we can use anyways to give people an idea on how much the site changes from release to release.

**Pull request merged**